core/src/main/java/org/testcontainers/containers/ContainerLifecycleHooks.java

```diff
@@ -0,0 +1,11 @@
+package org.testcontainers.containers;
+
+interface ContainerLifecycleHooks {
+    ContainerLifecycleHooks EMPTY = new ContainerLifecycleHooks() {};
+
+    default void configure() {}
```

Comment:
I had to wrap my head around, that `configure()` would generally be executed on a `BaseContainerDef` object, and the other hooks on `StartedContainer`. But I think this distinction makes a lot of sense and is good. However, that also indicates to me, that having this single interface include _all_ of those hooks is slightly wrong.